### PR TITLE
Unit test for the fix of NuGet/Home#1300.

### PR DESCRIPTION
**Pull request by feiling**
@deepakaravindr @emgarten @yishaigalatzer @zhili1208

### CODE REVIEW

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 1]
```diff
@@ -892,5 +893,129 @@ public void InstallCommand_DependencyResolutionFailure()
...
+                        configFileName
+                    };
+                    var result = CommandRunner.Run(
+                        nugetexe,
+                        Directory.GetCurrentDirectory(),
+                        string.Join(" ", args),
+                        true);
+
+                    // Assert
+                    Assert.True(credentialsPassedToRegistrationEndPoint);
```
Comment (feiling) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs:1006]: Without the fix, credentialsPassedToRegistrationEndPoint is false.

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 2]
```diff
@@ -892,5 +893,129 @@ public void InstallCommand_DependencyResolutionFailure()
...
+                ""@vocab"": ""http://schema.nuget.org/services#"",
+                ""comment"": ""http://www.w3.org/2000/01/rdf-schema#comment""
+                    }}");
+
+                    serverV3.Get.Add("/a/b/c/index.json", r =>
+                    {
+                        var h = r.Headers["Authorization"];
+                        var credential = String.IsNullOrEmpty(h) ? 
+                            null :
+                            System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
```
Comment (yishaigalatzer) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs:921]: use a using statement

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs [hunk 3]
```diff
@@ -892,5 +893,129 @@ public void InstallCommand_DependencyResolutionFailure()
...
+
+                        if (StringComparer.OrdinalIgnoreCase.Equals("test_user:test_password", credential))
+                        {
+                            return new Action<HttpListenerResponse>(response =>
+                            {
+                                response.StatusCode = (int)HttpStatusCode.OK;
+                                response.ContentType = "text/javascript";
```
Comment (yishaigalatzer) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: application/json ?

**Comment by emgarten** [comment]
Use the helper methods in the utility class for creating index.json

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 3 above)
Comment (feiling) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: This was copied from another test. In fact the content type is never checked by our client. Will change this.

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 2 above)
Comment (feiling) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs:921]: using on which object?

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 3 above)
Comment (emgarten) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: Change it in the other test v3 test also

**Comment by emgarten** [comment]
There is a utility method for creating the registration index.json entry also. It would be nice to keep the tests consistent.
:watch:

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 3 above)
Comment (yishaigalatzer) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs]: The other test is wrong. We should either remove the content type or use the correct one

Sent from my Windows Phone

File: test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs (hunk 2 above)
Comment (yishaigalatzer) [inline test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs:921]: System.text

Sent from my Windows Phone

**Comment by feiling** [comment]
@emgarten fixed. Util.AddRegistrationResource is now used.

**Comment by emgarten** [comment]
:shipit:

**Comment by feiling** [comment]
Merged

**Pull request closed without merging**